sdk/program/src/sysvar/slot_hashes.rs

```diff
@@ -97,30 +100,50 @@ impl SlotHashesSysvar {
     /// Get the position of an entry in the sysvar by its key.
     /// Returns `None` if the key is not found.
     pub fn position(slot: &Slot) -> Result<Option<usize>, ProgramError> {
-        get_pod_slot_hashes().map(|pod_hashes| {
+        Self::get_pod_slot_hashes().map(|pod_hashes| {
             pod_hashes
                 .binary_search_by(|PodSlotHash { slot: this, .. }| slot.cmp(this))
                 .ok()
         })
     }
-}
-
-fn get_pod_slot_hashes() -> Result<Vec<PodSlotHash>, ProgramError> {
-    let mut pod_hashes = vec![PodSlotHash::default(); MAX_ENTRIES];
-    {
-        let data = bytemuck::try_cast_slice_mut::<PodSlotHash, u8>(&mut pod_hashes)
-            .map_err(|_| ProgramError::InvalidAccountData)?;
 
-        // Ensure the created buffer is aligned to 8.
-        if data.as_ptr().align_offset(8) != 0 {
-            return Err(ProgramError::InvalidAccountData);
+    /// Return the slot hashes sysvar as a vector of `PodSlotHash`.
+    pub fn get_pod_slot_hashes() -> Result<Vec<PodSlotHash>, ProgramError> {
```

Comment:
I'd rather move toward dropping the `get_` so this aligns with the `SlotHashes` API. What do you think?

https://github.com/anza-xyz/agave/blob/efdbdc00cea2b89338b2e8e1f9ebb6ca77c8b5fa/sdk/program/src/slot_hashes.rs#L61-L63

Reply:
Does that mean that people will call `slot_hashes.slot_hashes()`? I'm not a fan to be honest :sweat_smile: -- how about something a bit more Rusty and go with `as_slice()`? https://doc.rust-lang.org/std/vec/struct.Vec.html#method.as_slice

Reply:
> Does that mean that people will call `slot_hashes.slot_hashes()`? I'm not a fan to be honest 😅

No, since the new function in question is not `&self`. However, that's exactly how the existing API works.

I'm suggesting `SlotHashesSysvar::pod_slot_hashes()`. I don't like `.as_slice()` since it adds a level of indirection on `SlotHashesSysvar` that's not super clear. If we had something like this it would make more sense to use `.as_slice()`:

```rust
struct SlotHashesSysvar(Vec<PodSlotHash>);
// or
struct PodSlotHashes(Vec<PodSlotHash>);
```

For those reasons I still think `SlotHashesSysvar::pod_slot_hashes()` is best.

Reply:
As a follow-up on your original suggestion for `get()` and `get_all()`, I think the problem here is that they're not just singular/plural versions of each other, but `SlotHashesSysvar::pod_slot_hashes()` actually returns a collection of `PodSlotHash`, where `get()` is just a single `Hash` (not `PodSlotHash`).

Reply:
Ah I see, my mistake, I misunderstood and thought you wanted to create a wrapper struct. In that case, I think `get_pod_slot_hashes()` is best, that way we still follow the `get` convention as with the other sysvars, like `Clock::get()`. `pod_slot_hashes()` may make it less clear that there's a syscall happening underneath

Reply:
Renamed back to `get_pod_slot_hashes`.